src/flamenco/vm/fd_vm.c

```diff
@@ -408,10 +408,22 @@ fd_vm_validate( fd_vm_t const * vm ) {
     case FD_INVALID: default: return FD_VM_ERR_INVALID_OPCODE;
     }
 
-    if( FD_UNLIKELY( instr.src_reg>10 ) ) return FD_VM_ERR_INVALID_SRC_REG; /* FIXME: MAGIC NUMBER */
-
-    int is_invalid_dst_reg = instr.dst_reg > ((validation_code == FD_CHECK_ST) ? 10 : 9); /* FIXME: MAGIC NUMBER */
-    if( FD_UNLIKELY( is_invalid_dst_reg ) ) return FD_VM_ERR_INVALID_DST_REG;
+    /* Check registers
+       https://github.com/solana-labs/rbpf/blob/v0.8.5/src/verifier.rs#L177 */
+
+    /* Source register */
+    if( FD_UNLIKELY( instr.src_reg>10 ) ) return FD_VM_ERR_INVALID_SRC_REG;
+
+    /* Special R11 register */
+    if( instr.dst_reg==11U
+        && FD_VM_SBPF_DYNAMIC_STACK_FRAMES( sbpf_version )
+        && instr.opcode.raw == 0x07
+        && ( instr.imm % FD_VM_SBPF_DYNAMIC_STACK_FRAMES_ALIGN )==0 )
```

Comment:
Just so I understand, this check isn't in Agave yet, but will be added? Might be worth adding a TODO to go back and check that the implementation matches theirs once they have it.

Reply:
the align is not in agave yet but they're adding it before release.